chain/processor.go

```diff
@@ -49,11 +49,11 @@ func (p *Processor) Prefetch(ctx context.Context, db Database) {
 		defer span.End()
 
 		// Store required keys for each set
-		alreadyFetched := make(map[string]*fetchData, len(p.blk.GetTxs()))
+		alreadyFetched := make(map[tstate.Key]*fetchData, len(p.blk.GetTxs()))
 		for _, tx := range p.blk.GetTxs() {
-			storage := map[string][]byte{}
+			storage := map[tstate.Key][]byte{}
 			for _, k := range tx.StateKeys(sm) {
-				sk := string(k)
+				sk := tstate.ToStateKeyArray(k)
```

Comment:
I wonder if the best middle-ground is just to use the approach in `BenchmarkOptimizedStringKeyed`?

https://pthevenet.com/posts/programming/go/bytesliceindexedmaps/

We then get most of the benefit while retaining flexibility 🤷 .